### PR TITLE
Breaking: use VError base signatures. migrate jscs to eslint. 

### LINKED ISSUE #76: Use VError && VError.info under the hood
The `context` option was implemented before verror's `info` option was available. Now that it's available upstream we should try and use that under the hood instead. Some implications in terms of how to access the context property and inspect it (verror uses a static method `verror.info()`). Should be able to continue to use the same serializer, however, since it already supports verror's `info` field. 

Replaces #55

### PR DESCRIPTION
**Pull request by DonutEspresso**
Fixes #76, #77.

This jettisons any custom restify-error constructor APIs in favor of using VError's constructor APIs. This also eliminates the custom `.context` property in favor of using VError's `info` construct. A `context` getter remains on all restify-error base classes to facilitate with migration. More details in README, but this should essentially make restify-errors completely compatible with the VError static methods.

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -13,71 +23,48 @@ var assert = require('assert-plus');
...
-        }
-        // 3
-        // otherwise, pass through to WError
-
-        return {
-            args: args,
-            options: options
-        };
+        // reconstruct verror ctor options from the cleaned up options
+        verrorArgs = [ verrorOpts ].concat(_.tail(args));
```
Comment (retrohacker) [inline lib/helpers.js]: Why is `verrorArgs` not just `verrorOpts`? Doesn't adding `args` back to `verrorOpts` result in the overlap of the sets being included in the array twice?

File: lib/makeConstructor.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @return {Function} a constructor function
+ */
+function makeConstructor(name, defaults) {
+
+    assert.string(name, 'name');
+    assert.optionalObject(defaults, 'defaults');
+
+    // code property doesn't have 'Error' in it. remove it.
+    var defaultCode = name.replace(new RegExp('[Ee]rror$'), '');
+    var d = _.assign({}, {
```
Comment (hekike) [inline lib/makeConstructor.js]: I know it's not in the linter but I prefer avoiding one letter variables as it's hard to know what they mean and also hard to search and replace them.

File: README.md [hunk 3]
```diff
@@ -6,12 +6,11 @@
 [![Dependency Status](https://david-dm.org/restify/errors.svg)](https://david-dm.org/restify/errors)
 [![devDependency Status](https://david-dm.org/restify/errors/dev-status.svg)](https://david-dm.org/restify/errors#info=devDependencies)
 [![bitHound Score](https://www.bithound.io/github/restify/errors/badges/score.svg)](https://www.bithound.io/github/restify/errors/master)
-[![NSP Status](https://img.shields.io/badge/NSP%20status-no%20vulnerabilities-green.svg)](https://travis-ci.org/restify/errors)
 
 > A collection of HTTP and REST Error constructors.
 
-The constructors can be used to new up Error objects with default status codes
-set.
+This module ships with a set of constructors that can be used to new up Error
```
[the remark this answers is not in the document]
Comment (DonutEspresso) [reply README.md]: `new Error`, can rephrase if confusing

File: lib/helpers.js (hunk 1 above)
Comment (DonutEspresso) [reply lib/helpers.js]: VError has various signatures, the one we care about is `new VError({}, 'sprintf', 'sprintf args' ...)`

restify-errors options can be appended to the first arg - so what we do here is strip out restify-error specific options from `opts`, rename the options leftover as `verrorOpts`, then concat with everything but the first argument of raw `args` (the first argument being itself).

File: lib/makeConstructor.js (hunk 2 above)
Comment (DonutEspresso) [reply lib/makeConstructor.js]: Good call, will rename

File: .eslintrc [hunk 4]
```diff
@@ -1,14 +1,16 @@
 {
+    // eslint: recommended automatically enables most/all rules from the
```
[the remark this answers is not in the document]
Comment (DonutEspresso) [reply .eslintrc]: :+1: done, JSCS was freezing/failing for unknown reasons, figured no better time to swap it out.

**Comment by DonutEspresso** [comment]
Comments addressed, once more?

**Comment by DonutEspresso** [comment]
Thanks all. Excited to get this in, this has been a long time coming.

**Pull request merged**